### PR TITLE
prepare provider for tfplugindocs

### PR DESCRIPTION
**Pull request by maxim-nazarenko**
Part of #82

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -7,20 +13,28 @@ The provider needs to be configured with the proper credentials before it can be
...
+## Schema
+
+### Optional
+
+- `ca_certificate` (String) Path to MikroTik's certificate authority
+- `host` (String) Hostname of the MikroTik router
+- `insecure` (Boolean) Insecure connection does not verify MikroTik's TLS certificate
+- `password` (String) Password for MikroTik api
+- `tls` (Boolean) Whether to use TLS when connecting to MikroTik or not
+- `username` (String) User account for MikroTik api
```
Comment (ddelnano) [inline docs/index.md:40]: username, password and host are not optional fields. So it seems like the provider's arguments aren't set correctly?

File: mikrotik/provider.go [hunk 2]
```diff
@@ -2,32 +2,50 @@ package mikrotik
...
+		if s.Default != nil {
+			if s.Default == "" {
+				desc += " Default: `\"\"`."
+			} else {
+				desc += fmt.Sprintf(" Default: `%v`.", s.Default)
+			}
+		}
+
+		return strings.TrimSpace(desc)
+	}
```
Comment (ddelnano) [inline mikrotik/provider.go:26]: It looks like this is used in the other PRs (#84, #85, #86, etc). Can we make this a function that each of the resources can share?

```go
func buildSchemaDescriptionBuilder(s *schema.Schema) string {
		desc := s.Description
		// add default value in description, if it was declared in the resource's schema.
		if s.Default != nil {
			if s.Default == "" {
				desc += " Default: `\"\"`."
			} else {
				desc += fmt.Sprintf(" Default: `%v`.", s.Default)
			}
		}

		return strings.TrimSpace(desc)
	}

func init() {
	schema.SchemaDescriptionBuilder = buildSchemaDescriptionBuilder
}
```

**Comment by ddelnano** [comment]
I'd like to merge this PR first so we can rebase the others to remove the shared changes. I had one question about why everything is considered optional, otherwise this looks good.

File: docs/index.md (hunk 1 above)
Comment (maxim-nazarenko) [reply docs/index.md:40]: that's how it is [declared](https://github.com/ddelnano/terraform-provider-mikrotik/blob/543dbda3b40aec544fcad8adad77a344b9fa7cfe/mikrotik/provider.go#L22):
```go
"username": {
    Type:        schema.TypeString,
    Required:    true,
    DefaultFunc: schema.EnvDefaultFunc("MIKROTIK_USER", nil),
    Description: "User account for mikrotik api",
},
```
so yes, it should either be set to `Optional` or update the docs and mention that there is a fallback to env vars.

**Comment by maxim-nazarenko** [comment]
@ddelnano we can turn those fields into `Required` - the will be fetched from env if user does not provide them in configuration.

**Comment by ddelnano** [comment]
I see. I didn't realize that having the env in use would make they optional.

**Comment by ddelnano** [comment]
@maxim-nazarenko please rebase the remaining doc update PRs and then we can get those merged.

**Comment by maxim-nazarenko** [comment]
@ddelnano I just relaised what you was referring to. 
It was an [issue](https://github.com/hashicorp/terraform-plugin-docs/pull/163) in `tfplugindocs` which rendered fields in `optional` section regardless of `Required: true` attribute.
I'll update it in follow-up PR.

Sharp eye!

**Pull request merged**